### PR TITLE
Admin login refactoring

### PR DESCRIPTION
**Pull request by davisvagris**

### CODE REVIEW

File: src/Services/Authentication/SecurityStrategy.php [hunk 1]
```diff
@@ -8,12 +8,12 @@
...
 {
     /**
-     * @param array $credentials
+     * @param \Cartalyst\Sentinel\Users\UserInterface|array|null  $credentials
      * @param bool $remember
      * @param bool $login
      * @return Reply
      */
-    public function authenticate( array $credentials, $remember = false, $login = true ): Reply;
+    public function authenticate( $credentials, $remember = false, $login = true ): Reply;
```
Comment (exabyssus) [inline src/Services/Authentication/SecurityStrategy.php]: Interface or array? Seems wrong

Comment (davisvagris) [reply src/Services/Authentication/SecurityStrategy.php]: It is weird, but sentinel allows

https://github.com/cartalyst/sentinel/blob/v2.0.17/src/Sentinel.php#L284

File: src/Services/Authentication/SessionSecurityService.php [hunk 2]
```diff
@@ -24,18 +24,21 @@ public function __construct( Sentinel $sentinel )
     }
 
     /**
-     * @param array $credentials
+     * @param \Cartalyst\Sentinel\Users\UserInterface|array|null  $credentials
```
Comment (rudolfsp1) [inline src/Services/Authentication/SessionSecurityService.php]: Kas tie par return types?  Kā tas vispār var būt, kad atgriež user vai masīvu? :D

File: src/Services/Authentication/SessionSecurityService.php [hunk 3]
```diff
@@ -24,18 +24,21 @@ public function __construct( Sentinel $sentinel )
...
      * @param boolean $remember
      * @param boolean $login
      * @return Reply
      */
-    public function authenticate( array $credentials, $remember = false, $login = true ): Reply
+    public function authenticate( $credentials, $remember = false, $login = true ): Reply
```
Comment (exabyssus) [inline src/Services/Authentication/SessionSecurityService.php]: Could be better with two methods authenticateUser authenticateWithCredentials or smth like that. Single responsibility principle

File: src/Http/Controllers/Admin/UsersController.php [hunk 4]
```diff
@@ -63,6 +63,10 @@ protected function form(Form $form)
         $model = $form->getModel();
 
         $form->addEventListener('create.before', function () use ($model) {
+            if (empty($model->provider)) {
```
Comment (exabyssus) [inline src/Http/Controllers/Admin/UsersController.php]: Is this necessary? If DB has this as default value

File: src/Http/Controllers/Admin/SecurityController.php [hunk 5]
```diff
@@ -79,4 +87,28 @@ public function postLogout()
...
+    /**
+     * @param \Illuminate\Http\Request $request
+     *
+     * @return \Cartalyst\Sentinel\Users\UserInterface|array|null
+     */
+    protected function findAuthUser( Request $request )
+    {
+        $credentials = Arr::only( $request->get( 'user' ), [ 'email', 'password' ] );
+    
+        return Sentinel::findByCredentials(array_merge($credentials, [
```
Comment (exabyssus) [inline src/Http/Controllers/Admin/SecurityController.php]: What about existing users? It this tested and backwards compatible?

Comment (davisvagris) [reply src/Http/Controllers/Admin/SecurityController.php]: Yes! Migration has default value!

File: src/Services/Authentication/SessionSecurityService.php [hunk 6]
```diff
@@ -23,19 +23,44 @@ public function __construct( Sentinel $sentinel )
...
-
-        if( $user )
+        if ($credentials)
         {
-            return new SuccessReply( trans( 'auth.success' ) );
+            $user = $this->sentinel->authenticate( $credentials, $remember, $login );
+    
+            if ($user)
+            {
+                return new SuccessReply(trans('auth.success'));
```
Comment (rudolfsp1) [inline src/Services/Authentication/SessionSecurityService.php]: Coming from a person, who has no idea why this was named this way, this looks very weird. Reply and response are similar but not the same. As far as I know, people usually choose the word "response" when using it in software development context.

https://english.stackexchange.com/questions/48385/difference-between-response-and-reply

Comment (exabyssus) [reply src/Services/Authentication/SessionSecurityService.php]: SuccessReply was implemented in 2016, not with this PR therefore it's out of scope. Nonetheless, good point.

**Comment by exabyssus** [comment]
I don't think that this is a good upgrade for Authentification. We should make a port-and-adapter style Auth.

**Pull request closed without merging**